src/server/utils/WireGuard.ts

```diff
@@ -0,0 +1,394 @@
+    for (let i = 2; i < 255; i++) {
+      const client = Object.values(config.clients).find((client) => {
+        return client.address === WG_DEFAULT_ADDRESS.replace('x', i.toString());
+      });
+
+      if (!client) {
+        address = WG_DEFAULT_ADDRESS.replace('x', i.toString());
+        break;
+      }
+    }
+
+    if (!address) {
+      throw new Error('Maximum number of clients reached.');
+    }
+
+    // Create Client
+    const id = crypto.randomUUID();
+    const client = {
+      id,
+      name,
+      address,
+      privateKey,
+      publicKey,
+      preSharedKey,
+
+      createdAt: new Date().toISOString(),
+      updatedAt: new Date().toISOString(),
+
+      enabled: true,
+    };
+
+    config.clients[id] = client;
+
+    await this.saveConfig();
+
+    return client;
+  }
+
+  async deleteClient({ clientId }: { clientId: string }) {
+    const config = await this.getConfig();
+
+    if (config.clients[clientId]) {
+      // eslint-disable-next-line @typescript-eslint/no-dynamic-delete
+      delete config.clients[clientId];
+      await this.saveConfig();
+    }
+  }
+
+  async enableClient({ clientId }: { clientId: string }) {
+    const client = await this.getClient({ clientId });
+
+    client.enabled = true;
+    client.updatedAt = new Date().toISOString();
+
+    await this.saveConfig();
+  }
+
+  async disableClient({ clientId }: { clientId: string }) {
+    const client = await this.getClient({ clientId });
+
+    client.enabled = false;
+    client.updatedAt = new Date().toISOString();
+
+    await this.saveConfig();
+  }
+
+  async updateClientName({
+    clientId,
+    name,
+  }: {
+    clientId: string;
+    name: string;
+  }) {
+    const client = await this.getClient({ clientId });
+
+    client.name = name;
+    client.updatedAt = new Date().toISOString();
+
+    await this.saveConfig();
+  }
+
+  async updateClientAddress({
+    clientId,
+    address,
+  }: {
+    clientId: string;
+    address: string;
+  }) {
+    const client = await this.getClient({ clientId });
+
+    if (!isValidIPv4(address)) {
+      throw new ServerError(`Invalid Address: ${address}`, 400);
+    }
+
+    client.address = address;
+    client.updatedAt = new Date().toISOString();
+
+    await this.saveConfig();
+  }
+
+  async __reloadConfig() {
+    await this.__buildConfig();
+    await this.__syncConfig();
+  }
+
+  async restoreConfiguration(config: string) {
+    debug('Starting configuration restore process.');
+    // TODO: sanitize config
+    const _config = JSON.parse(config);
+    await this.__saveConfig(_config);
+    await this.__reloadConfig();
+    debug('Configuration restore process completed.');
+  }
+
+  async backupConfiguration() {
+    debug('Starting configuration backup.');
+    const config = await this.getConfig();
+    const backup = JSON.stringify(config, null, 2);
+    debug('Configuration backup completed.');
+    return backup;
+  }
+
+  // Shutdown wireguard
+  async Shutdown() {
```

Comment:
this method is not called anywhere

Reply:
What's fixed here? It's not a proper use of the Layouts folder.
See: https://github.com/nuxt-ui-pro/saas/tree/main/app/layouts